xla/tests/onednn_layer_norm_test.cc

```diff
@@ -141,6 +141,75 @@ TEST_F(LayerNormTest, LayerNormTest0_F16) {
   MatchOptimizedHlo(layer_norm_module_str, onednn_layer_norm_);
 }
 
+// Test case encountered in models like TFViTForImageClassification in
+// HuggingFace
+// (https://huggingface.co/docs/transformers/model_doc/vit#transformers.TFViTForImageClassification)
+TEST_F(LayerNormTest, LayerNormTest1_BF16) {
+  const char* layer_norm_module_str = R"(
+  HloModule layer_norm.test, entry_computation_layout={(bf16[160,197,768]{2,1,0}, bf16[768]{0}, bf16[768]{0})->bf16[160,197,768]{2,1,0}}
+  region_add {
+    Arg_0.7555 = f32[] parameter(0)
+    Arg_1.7556 = f32[] parameter(1)
+    ROOT add.7557 = f32[] add(Arg_0.7555, Arg_1.7556)
+  }
+  ENTRY main {
+    Arg_0.1 = bf16[160,197,768]{2,1,0} parameter(0), sharding={replicated}
+    Arg_0.2 = bf16[768]{0} parameter(1), sharding={replicated}
+    Arg_0.3 = bf16[768]{0} parameter(2), sharding={replicated}
+    convert.80 = f32[160,197,768]{2,1,0} convert(Arg_0.1)
+    constant.81 = f32[] constant(0)
+    convert.82 = f32[] convert(constant.81)
+    reduce.87 = f32[160,197]{1,0} reduce(convert.80, convert.82), dimensions={2}, to_apply=region_add
+    constant.88 = s32[] constant(768)
+    convert.89 = f32[] convert(constant.88)
```

Comment:
Can we just make `constant.88` f32?
```suggestion
    constant.88 = f32[] constant(768)
```

Reply:
Modifying constant to fp32 won't capture the exact HLO generated when running models, This test captures the exact case for reduce sum, coming from keras.LayerNormalization. As the divisor constant for mean calculation is always created as s32, we also expect the pattern to be s32 constant ->convert->broadcast.
Also we also observe the same HLO for other tests as well in this file.

Reply:
I understand. But this means we'll have to update the test each time keras or XLA CPU passes make any changes that affect the HLO though.